haskell/src/Lentille/GitHub/User.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-overlapping-patterns #-}
+
+module Lentille.GitHub.User where
+
+import Data.Morpheus.Client
+import Lentille
+import Lentille.GitHub.RateLimit (retryCheck)
+import Lentille.GraphQL
+import Monocle.Prelude
+
+newtype DateTime = DateTime Text deriving (Show, Eq, EncodeScalar, DecodeScalar)
+
+data IdentInfo = IdentInfo
+  { iiName :: Maybe Text,
+    iiCompany :: Maybe Text
+  }
+
+defineByDocumentFile
+  ghSchemaLocation
+  [gql|
+    query getUser ($login: String!) {
+      rateLimit {
+        used
+        remaining
+        resetAt
+      }
+      user(login: $login) {
+        name
+        company
+      }
+    }
+  |]
+
+transformResponse :: GetUser -> (Maybe RateLimit, IdentInfo)
```

Comment:
```suggestion
transformResponse :: GetUser -> (RateLimit, Maybe IdentInfo)
```

Reply:
Yes, I've removed the Maybe RateLimit. However we should get a IdentInfo or an exception will be raised, so for me that's an IdentInfo but not a Maybe IdentInfo.